nncf/openvino/graph/node_utils.py

```diff
@@ -325,15 +325,15 @@ def get_weight_channel_axes(node: NNCFNode, weights_port_id: int) -> List[int]:
     if node.metatype == OVMatMulMetatype:
         assert isinstance(node.layer_attributes, OVLayerAttributes)
         assert len(channel_axes) == 1
-        assert channel_axes[0] in [-1, -2]
-        const_attrs = node.layer_attributes.constant_attributes[weights_port_id]
         matmul_channel_axis = channel_axes[0]
-        if const_attrs["transpose"]:
-            transpose_swap = {-1: -2, -2: -1}
-            matmul_channel_axis = transpose_swap[matmul_channel_axis]
-        shape = node.layer_attributes.constant_attributes[weights_port_id]["shape"]
-        matmul_channel_axis = len(shape) + matmul_channel_axis
-        channel_axes = list(range(len(shape) - 2))
-        channel_axes.append(matmul_channel_axis)
+        const_attrs = node.layer_attributes.constant_attributes[weights_port_id]
```

Comment:
IMO, it would be really helpful to add some comments about the calculations below. With samples, if possible.

Reply:
Honestly, I could not find a better idea than copy test cases here, but the comment become very huge. I hope, added tests are enough for now.